data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java

```diff
@@ -0,0 +1,193 @@
+
+public class TmdbDataRetriever {
+
+    private List<MovieInfo> tmdbMovies;
+    private final static String API_KEY = "9eda0433936b655a246eef78d367b530";
+    private HttpClient httpClient;
+    private HttpTools httpTools;
+    private TmdbMovies moviesInstance;
+    private TmdbSearch searchInstance;
+
+    public TmdbDataRetriever() {
+        httpClient = new SimpleHttpClientBuilder().build();
+        httpTools = new HttpTools(httpClient);
+        moviesInstance = new TmdbMovies(API_KEY, httpTools);
+        searchInstance = new TmdbSearch(API_KEY, httpTools);
+    }
+
+    public Movie getMovie(Movie movie) throws MovieDbException {
+        List<MovieInfo> movies = searchInstance.searchMovie(movie.getName(), 1, "de-DE", false, 0, 0, SearchType.NGRAM)
+                .getResults();
+
+        if (movies.size() != 0) {
+            MovieInfo mi = moviesInstance.getMovieInfo(movies.get(0).getId(), "de-DE", "");
+            // MovieInfo mi = moviesInstance.getMovieInfo(550, "de-DE",
+            // "RELEASES");
+            movie = new Movie();
+            movie.setName(mi.getTitle());
+            movie.setDescription(mi.getOverview());
+            movie.setTmdbId(mi.getId());
+            movie.setStudio(getStudio(mi));
+            // m.setCover(cover);
+
+            // movie.setLengthMinutes(mi.getRuntime()); Can't use this one cause
+            // it seems to be broken (returns allways 0)
+            movie.setLengthMinutes(mi.getRuntime());
+            movie.setAgeControl(getAgeControl(mi));
+            movie.setGenre(getGenre(mi));
+            movie.setAuthor(getAuthor(mi));
+            movie.setDirector(getDirector(mi));
+
+        } else {
+            throw new MovieDbException(null, "Movie not found");
+        }
+
+        return movie;
+    }
+
+    private String getStudio(MovieInfo mi) {
+        String result = "";
+        List<ProductionCompany> list = mi.getProductionCompanies();
+        if (list.size() > 0) {
+            Iterator<ProductionCompany> i = list.iterator();
+            while (i.hasNext()) {
+                ProductionCompany item = i.next();
+                result += item.getName() + (i.hasNext() ? ", " : "");
+            }
+            return result;
+        }
+        return "Keine Angabe";
+    }
+
+    private String getAuthor(MovieInfo mi) throws MovieDbException {
+        List<MediaCreditCrew> crew = moviesInstance.getMovieCredits(mi.getId()).getCrew();
+
+        String result = "";
+
+        List<MediaCreditCrew> list = crew.stream()
+                .filter(c -> c.getDepartment().equals("Writing") && c.getJob().equals("Screenplay"))
+                .collect(Collectors.toList());
+
+        if (list.size() > 0) {
+            Iterator<MediaCreditCrew> i = list.iterator();
+            while (i.hasNext()) {
+                MediaCreditCrew item = i.next();
+                result += item.getName() + (i.hasNext() ? ", " : "");
+            }
```

Comment:
Could we extract something like a "concatStrings" method here? This looks like a lot of duplicate code across the whole file